### PR TITLE
Loosen the rack requirement to allow for testing with upcoming rack 2

### PR DESCRIPTION
**Pull request by twalpole**
Rails master has just moved to requiring the rack master branch which is working on rack 2.0 - This loosens up omniauths requirement to allow for testing and development. I specifically need it for testing devise with rails master

### CODE REVIEW

**Comment by pwnall** [comment]
This is biting me too. Is there anything that I can do to help get this PR in? 

Is any part of it concerning? If so, I could try to come up with a different approach.

Thank you!

File: lib/omniauth/builder.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@ def rack14?
       Rack.release.split('.')[1].to_i >= 4
```
Comment (md5) [inline lib/omniauth/builder.rb]: It looks like this should be tightened up to check that the first component of the version number is `1`.

**Comment by twalpole** [comment]
@md5 ok I fixed that check - 4 months ago when I created this PR all the tests passed correctly, now all the ruby 1.8.7 tests fail.  Obviously some gem has been updated to only support ruby 1.9.3+ and the gem restrictions in this project need to be updated if 1.8.7 support is still considered needed.  That being said, there has been no official response to this PR yet so I'm reticent to put in more work figuring out which gem(s) are the issue until there is at least some kind of response

**Comment by md5** [comment]
@twalpole I totally understand regarding the Ruby 1.8.7 thing. I ran into it with #829 as well.

**Comment by sferik** [comment]
Thanks!

**Comment by md5** [comment]
:+1:

**Comment by pwnall** [comment]
Thank you!

**Pull request merged**